Voice interactions with Gemini Live and ros-mcp-server added to Gemini example.

tracelarue opened the pull request:
**Gemini Live for low-latency bidirectional voice interactions with ros-mcp-server.** 

- Added gemini_live to examples/2_gemini.
- Enables audio input from the user and audio output from Gemini.
- Enables Gemini Live to use ros-mcp-server.
- Tested in ubuntu 22.04, python 3.10, ros2 humble.

stex2005 commented on examples/2_gemini/gemini_live/mcp_config.json:
We can avoid having this file here, as it is a duplicate of main installation. 
I recommend removing it and suggesting users to create their own as they will have to modify it anyway.

stex2005 commented on examples/2_gemini/gemini_live/pyproject.toml:
I would avoid creating a project inside the repository, but since this is a client project, it makes sense. Maybe we should move this example into a clients folder

tracelarue replied:
Removed pyproject.toml and uv.lock in favor of uv pip install dependencies in the readme.

stex2005 commented:
<img width="976" height="1152" alt="image" src="https://github.com/user-attachments/assets/754d1c79-7af4-44d8-b586-ea674c385163" />

Raise this error during installation, seems that I need a system-package: portaudio.

stex2005 commented:
@tracelarue @rjohn-v Another good next step would be to provide a `dockerized` version of the Gemini client, so it can be run more easily in different environments. Since the client is only a tool within this project, I don’t think we should invest too much effort in tightly integrating it into the repo. A simplified version of `client_gemini` (without audio support) would already be a good, lightweight solution and would find a good place in `clients` folder.

mokcontoro commented:
@tracelarue wow, voice command sounds super cool. thanks for your contributions. I cannot wait for trying this feature soon!

tracelarue commented on examples/2_gemini/gemini_live/README.md:
```diff
@@ -0,0 +1,100 @@
+
+**Pre-requisites** See the [installation instructions](../../../docs/installation.md) for detailed setup steps.
+
+**Tested In:** Ubuntu 22.04, Python 3.10, ROS2 Humble
```
I am unsure, I have not used WSL before.

tracelarue replied:
Ok, I'll specify that for now. I'll work on learning WSL so I can support it in the future.

tracelarue replied:
May work on WSL after the changes removing the project. I dont know WSL yet to be able to try it myself. I specified that it only works in ubuntu for now, but can update that if you find it now works in WSL.

tracelarue commented on examples/2_gemini/gemini_live/mcp_handler.py:
It served to connect Gemini to the mcp server. I have removed this and changed gemini_client.py to use the latest Google documentation for connecting to mcp servers, so this file is no longer needed.

tracelarue commented:
@stex2005 Ready for review

Changes made:
- removed mcp_config.json and instruct user to create their own
- removed pyproject.toml and uv.lock. Now uses uv pip install to install dependencies in the existing ros-mcp .venv
- updated README.md to specify .env location
- renamed gemini_live.py to gemini_client.py
- specified it only works on ubuntu in README.md (try WSL again after these changes)
- removed mcp_handler.py in favor of Google's latest method to connect to mcp servers
- Other gemini_client.py improvements and README.md updates

stex2005 commented:
> @stex2005 Ready for review
> 
> Changes made:
> 
> * removed mcp_config.json and instruct user to create their own
> * removed pyproject.toml and uv.lock. Now uses uv pip install to install dependencies in the existing ros-mcp .venv
> * updated README.md to specify .env location
> * renamed gemini_live.py to gemini_client.py
> * specified it only works on ubuntu in README.md (try WSL again after these changes)
> * removed mcp_handler.py in favor of Google's latest method to connect to mcp servers
> * Other gemini_client.py improvements and README.md updates

@tracelarue thank you for the contribution. I will try it :)

stex2005 commented:
@tracelarue thanks again for your contribution.

I've tried your example on WSL and Ubuntu and it still gives some compatibility issues due to hardware/driver changes on camera/microphone. I would suggest making this example a simplified version of the `gemini_client` that works only with text input/output. At this moment the example code is too broad and complex to be maintained.

I am working on this simplified example and I will soon share it.

tracelarue commented:
@stex2005 Thanks for the review. I’ll start working on a text-only version and test it on WSL for compatibility. Could you share your hardware setup and specify where the issue occurred? Was the gemini_client able to launch and connect to the MCP? I’ve previously run the live version on a Raspberry Pi, where I had to adjust the mic and speaker sampling rates for hardware compatibility.
